src/EA.Iws.RequestHandlers/NotificationMovements/BulkReceiptRecovery/ReceiptRecoveryRecoveryQuantityUnitRule.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace EA.Iws.RequestHandlers.NotificationMovements.BulkReceiptRecovery
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Core.Movement;
+    using Core.Movement.BulkReceiptRecovery;
+    using Core.Rules;
+    using Domain.Movement;
+    using Domain.NotificationApplication.Shipment;
+
+    public class ReceiptRecoveryRecoveryQuantityUnitRule : IReceiptRecoveryContentRule
+    {
+        private readonly IShipmentInfoRepository repo;
+
+        public ReceiptRecoveryRecoveryQuantityUnitRule(IShipmentInfoRepository repo)
+        {
+            this.repo = repo;
+        }
+
+        public async Task<ReceiptRecoveryContentRuleResult<ReceiptRecoveryContentRules>> GetResult(List<ReceiptRecoveryMovement> movements, Guid notificationId)
```

Comment:
You need to incorporate `ShipmentQuantityUnitsMetadata` within this. Basically, there's only one unit assigned to a notification, e.g. Tonnes. However, they will also be allowed to use Kilograms too. Have a look at the `PrenotificationQuantityUnitRule` for the prenotification.